packages/next/client/page-loader.js

```diff
@@ -70,14 +69,32 @@ export default class PageLoader {
         return
       }
 
-      // Load the script if not asked to load yet.
       if (!this.loadingRoutes[route]) {
-        this.loadScript(route)
+        // Make sure we don't load a dependency that's already loaded
+        const loadedModules = new Set(
+          Array.from(document.getElementsByTagName('script'), el => {
+            const results = /\/_next\/[^?]*/g.exec(el.src)
+            return results ? results[0] : el.src
+          })
+        )
+        const deps = this.getDependencies(route)
+        const urlsToLoad = deps.filter(url => !loadedModules.has(url))
+
         this.loadingRoutes[route] = true
+        urlsToLoad.forEach(url => this.loadScript(url, route))
+        this.loadRouteBundle(route)
       }
     })
   }
 
+  // Retrieve a list of dependencies for a given route from the build manifest
+  getDependencies (route) {
```

Comment:
It feels weird to have this in the class, given it's reading a global side-effect. Let's move it out to a top level function alongside `supportsPreload`.